metadata_backend/services/metax_service_handler.py

```diff
@@ -224,7 +225,13 @@ async def update_dataset_with_doi_info(self, datacite_info: Dict, _metax_ids: Li
 
             # Map fields from doi info to Metax schema
             mapper = MetaDataMapper(metax_id["schema"], metax_data["research_dataset"], datacite_info)
-            mapped_metax_data = mapper.map_metadata()
+            try:
+                mapped_metax_data = mapper.map_metadata()
+            except SubjectNotFoundException as error:
```

Comment:
Yea I originally didn't have this because I too thought that the subjects should always correspond with the schema. But then I figured you can technically still feed the method faulty values which would cause issues. Will it ever happen? Maybe not, but if it ever does, this avoids returning a 500 response.

Reply:
Yes, makes sense.